code/datums/status_effects/buffs.dm

```diff
@@ -366,24 +366,28 @@
 	alert_type = null
 	/// This diminishes the healing of fleshmend the higher it is.
 	var/tolerance = 1
+	/// This diminishes the healing of fleshmend if the user is cold when it is activated
+	var/freezing = FALSE
 	var/instance_duration = 10 // in ticks
 	/// a list of integers, one for each remaining instance of fleshmend.
 	var/list/active_instances = list()
 	var/ticks = 0
 
 /datum/status_effect/fleshmend/on_apply()
 	tolerance += 1
+	freezing = (owner.bodytemperature + 50 <= owner.dna.species.body_temperature)
 	active_instances += instance_duration
```

Comment:
Can you turn this code into a proc? So that the refresh and on_apply both use the same logic 100% of the time. Should make it less error-prone in the future